contracts/vesting/src/storage.rs

```diff
@@ -0,0 +1,181 @@
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct DistributionInfo {
+    pub start_timestamp: u64,
+    pub end_timestamp: u64,
+    pub amount: u128,
+}
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct VestingInfo {
+    pub amount: i128,
+    pub distribution_info: DistributionInfo,
+}
+
+impl MinterInfo {
+    pub fn get_curve(&self) -> Curve {
+        Curve::Constant(self.mint_cap)
+    }
+}
+
+impl DistributionInfo {
+    pub fn get_curve(&self) -> Curve {
+        Curve::SaturatingLinear(SaturatingLinear {
+            min_x: self.start_timestamp,
+            min_y: self.amount,
+            max_x: self.end_timestamp,
+            max_y: 0u128,
+        })
+    }
+}
+
+pub fn save_admin(env: &Env, admin: &Address) {
+    env.storage().persistent().set(&DataKey::Admin, admin);
+}
+
+pub fn get_admin(env: &Env) -> Address {
+    env.storage()
+        .persistent()
+        .get(&DataKey::Admin)
+        .unwrap_or_else(|| {
+            log!(&env, "Vesting: Get admin: Critical error - No admin found");
+            panic_with_error!(env, ContractError::NoAdminFound);
+        })
+}
+
+pub fn save_balance(env: &Env, address: &Address, balance: &i128) {
+    env.storage().persistent().set(address, balance);
+}
+
+pub fn save_vesting(env: &Env, address: &Address, vesting_info: &VestingInfo) {
+    env.storage().instance().set(address, vesting_info);
+}
+
+pub fn get_vesting(env: &Env, address: &Address) -> Result<VestingInfo, ContractError> {
+    let vesting_info = env.storage().instance().get(address).unwrap_or_else(|| {
+        log!(&env, "Vesting: Get vesting schedule: Critical error - No vesting schedule found for the given address");
+        panic_with_error!(env, ContractError::VestingNotFoundForAddress);
+    });
+
+    Ok(vesting_info)
+}
+
+pub fn remove_vesting(env: &Env, address: &Address) {
+    env.storage().instance().remove(address);
+}
+
+// TODO: uncomment when needed
+// pub fn get_allowances(env: &Env, owner_spender: &(Address, Address)) -> i128 {
+//     env.storage().persistent().get(owner_spender).unwrap_or_else(|| {
+//             log!(&env, "Vesting: Get allowance: Critical error - No allowance found for the given address pair");
+//             panic_with_error!(env, ContractError::AllowanceNotFoundForGivenPair);
+//         })
+// }
+
+// pub fn save_allowances(env: &Env, owner_spender: &(Address, Address), amount: i128) {
+//     env.storage().persistent().set(owner_spender, &amount);
+// }
+
+pub fn save_minter(env: &Env, minter: &MinterInfo) {
+    env.storage().persistent().set(&DataKey::Minter, minter);
+}
+
+pub fn get_minter(env: &Env) -> MinterInfo {
+    env.storage()
+        .persistent()
+        .get(&DataKey::Minter)
+        .unwrap_or_else(|| {
+            log!(
+                &env,
+                "Vesting: Get minter: Critical error - No minter found "
+            );
+            panic_with_error!(env, ContractError::MinterNotFound);
+        })
+}
```

Comment:
Let's ease this a little. Not having a minter is not a critical error, it's probably the base scenario.

```suggestion
pub fn get_minter(env: &Env) -> MinterInfo {
    env.storage()
        .persistent()
        .get(&DataKey::Minter)
        .unwrap_or_else(|| {
            MinterInfo {
               address: None,
               mint_cap: 0,
            }
        })
}
```

Reply:
kept it as:

```Rust
    env.storage()
        .persistent()
        .get(&DataKey::Minter)
```

and we check the value when needed